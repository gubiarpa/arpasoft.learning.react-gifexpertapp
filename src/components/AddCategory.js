import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  }

  const handleSubmit = (e) => {
    e.preventDefault(); // evita comportamiento por default
    if (inputValue.trim().length > 2) {
      setCategories(categories => [inputValue, ...categories]);
      setInputValue('');
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input
        type="text"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  );

  AddCategory.protoTypes = {
    setCategories: PropTypes.func.isRequired
  }

}