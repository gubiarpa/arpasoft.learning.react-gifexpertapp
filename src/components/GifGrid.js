import React, { useState, useEffect } from 'react';

export const GifGrid = ({category}) => {
  
  const [count, setCount] = useState(0);

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async() => {
    const url = `https://api.giphy.com/v1/stickers/search?q=Dragon Ball Z&limit=10&api_key=PFVvhfL0JApPVrMjn6jA3a73AnLeedP8`;
    const resp = await fetch(url);
    const { data } = await resp.json();

    const gifs = data.map( img => ({
      id: img.id,
      title: img.title,
      url: img.images?.downsized_medium.url
    }))

    console.log(gifs);
  }


  return (
    <>
      <h3> {category} </h3>
      <h3> {count} </h3>
      <button onClick={ () => setCount(count + 1) }></button>
    </>
  );
}