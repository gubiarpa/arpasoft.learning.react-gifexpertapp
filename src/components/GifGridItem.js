import React from 'react';

export const GifGridItem = ({title, url}) => {
  return (
    <div className="card animate__animated animate__fadeInUp">
      <img src={url} alt={title} className="animate__animated animate__fadeInUp" />
      <p>{title}</p>
    </div>
  );
}