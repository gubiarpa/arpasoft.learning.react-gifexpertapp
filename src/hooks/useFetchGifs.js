import { useState } from 'react';

export const useFetchGifs = () => {

  const [state, setState] = useState({
    data: [],
    loading: true
  });

  setTimeout(() => {
    setState({
      data: [2, 9, 0, 7],
      loading: false
    });
  }, 3000);

  return state;

}