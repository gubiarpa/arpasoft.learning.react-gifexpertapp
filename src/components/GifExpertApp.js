import React, { useState } from 'react';
import { AddCategory } from './AddCategory';

const GifExpertApp = () => {
  
  const [categories, setCategories] = useState(
    [
      'One Punch Man',
      'Samurai X',
      'Saint Seiya',
      'Dragon Ball Z'
    ]
  );

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        {
          categories.map( category => {
            return <li key={ category }> { category }</li>;
          })
        }
      </ol>
    </>
  );
}

export default GifExpertApp;