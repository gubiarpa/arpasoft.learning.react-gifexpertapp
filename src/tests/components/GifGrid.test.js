import React from 'react';
import '@testing-library/jest-dom';
import { GifGrid } from "../../components/GifGrid";
import { shallow } from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {

    const category = 'One Punch';

    test('Debe mostrar el componente correcto y hacer match con el snapshot', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot(); 
    });
    
    test('Debe mostrar items cuando se cargan imágenes', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https//localhost/cualquiera/huevada.jpg',
            title: 'Cualquier Huevada'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={ category } />);
        expect(wrapper).toMatchSnapshot(); 
    });
    
    test('Debe mostrar items cuando se cargan imágenes useFetchGifs', () => {
        const gifs = [];
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot(); 
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });

});