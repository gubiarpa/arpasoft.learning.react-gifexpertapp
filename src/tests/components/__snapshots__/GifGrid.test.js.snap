// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe mostrar el componente correcto y hacer match con el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInUp"
  >
     
    One Punch
     
  </h3>
  <p
    className="animate__animated animate__fadeInLeft"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> Debe mostrar items cuando se cargan imágenes 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInUp"
  >
     
    One Punch
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier Huevada"
      url="https//localhost/cualquiera/huevada.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid /> Debe mostrar items cuando se cargan imágenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInUp"
  >
     
    One Punch
     
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
